pageserver/src/tenant/storage_layer/layer.rs

```diff
@@ -1234,6 +1234,9 @@ impl LayerInner {
             // first we spawn off the async part, then it will spawn_blocking the blocking part
             Self::spawn(start_evicting.instrument(span));
         }
+
+        // if can_evict is now false, we must cancel the eviction but get_or_maybe_download has
+        // already done that.
```

Comment:
Not sure about this after all. It would be sensible to remove the whole wanted_evicted instead.

Reply:
Decided to remove `wanted_evicted: AtomicBool` altogether instead of worrying about this more: https://github.com/neondatabase/neon/pull/7030/commits/6c062d9edad001ec5f51c542de868677fdaf0bd0